pyro/infer/svgd.py

```diff
@@ -138,78 +130,91 @@ class SVGD(object):
         adam = Adam({"lr": 0.1})
         svgd = SVGD(model, kernel, adam, num_particles=50, max_plate_nesting=0)
 
-        def gradient_callback(squared_gradients):
-            print(squared_gradients)  # this helps us monitor convergence
-
         for step in range(500):
-            svgd.step(model_arg1, model_arg2, gradient_callback=gradient_callback)
+            svgd.step(model_arg1, model_arg2)
+
         final_particles = svgd.get_named_particles()
 
     References
 
     [1] "Stein Variational Gradient Descent: A General Purpose Bayesian Inference Algorithm,"
         Qiang Liu, Dilin Wang
+    [2] "Kernelized Complete Conditional Stein Discrepancy,"
+        Raghav Singhal, Saad Lahlou, Rajesh Ranganath
     """
-    def __init__(self, model, kernel, optim, num_particles, max_plate_nesting):
+    def __init__(self, model, kernel, optim, num_particles, max_plate_nesting, mode="univariate"):
         assert callable(model)
         assert isinstance(kernel, SteinKernel), "Must provide a valid SteinKernel"
         assert isinstance(optim, pyro.optim.PyroOptim), "Must provide a valid Pyro optimizer"
         assert num_particles > 1, "Must use at least two particles"
         assert max_plate_nesting >= 0
+        assert mode in ['univariate', 'multivariate'], "mode must be one of (univariate, multivariate)"
 
         self.model = vectorize(model, num_particles, max_plate_nesting)
         self.kernel = kernel
         self.optim = optim
         self.num_particles = num_particles
         self.max_plate_nesting = max_plate_nesting
+        self.mode = mode
 
         self.loss = Trace_ELBO().differentiable_loss
         self.guide = _SVGDGuide(self.model)
 
     def get_named_particles(self):
         """
-        Get a dictionary of named particles of the form {name: particle}
+        Get a dictionary of named particles of the form {name: particle}. The leading dimension
```

Comment:
nit: the plural in this name is confusing I would suggest replacing this comment with
```
Create a dictionary mapping name to vectorized value, of the form ``{name: tensor}``. The leading dimension of each tensor corresponds to particles. I.e. this creates a struct of arrays.
```

Reply:
done